File: myflix-core/src/main/java/com/maszlovicskrisztian/myflix_core/dtos/response/ShowDetailsResponse.java
package com.maszlovicskrisztian.myflix_core.dtos.response;

import com.maszlovicskrisztian.myflix_core.dtos.SeasonDetails;

import java.util.List;

public record ShowDetailsResponse(
        Long id,
        String title,
        String overview,
        String posterPath,
        String backdropPath,
        Integer seasonCount,
        Integer episodeCount,
        List<SeasonDetails> seasons,
        List<String> genres
) {
}

File: myflix-frontend/src/app/model/show-details-response.ts
import { SeasonDetails } from "./season-details";

export type ShowDetailsResponse = {
    id: number;
    title: string;
    overview: string;
    backdropPath: string | null;
    seasonCount: number;
    episodeCount: number;
    seasons: Array<SeasonDetails>;
    genres: Array<string>;
}